packages/shared/src/hooks/useChecklist.ts

```diff
@@ -0,0 +1,59 @@
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { ChecklistStepType } from '../lib/checklist';
+import { Action, ActionType } from '../graphql/actions';
+
+export type UseChecklistProps = {
+  steps: ChecklistStepType[];
+};
+
+export type UseChecklist = {
+  steps: ChecklistStepType[];
+  openStep: string | undefined;
+  onToggleStep: (action: Action) => void;
+  isDone: boolean;
+  activeStep: ActionType | undefined;
+  completedSteps: ChecklistStepType[];
+};
+
+const useChecklist = ({ steps }: UseChecklistProps): UseChecklist => {
+  const activeStep = useMemo(
+    () => steps.find((item) => !item.action.completedAt)?.action.type,
+    [steps],
+  );
+  const [openStep, setOpenStep] = useState<string>(activeStep);
+
+  useEffect(() => {
+    setOpenStep(activeStep);
+  }, [activeStep]);
```

Comment:
Why do you need both?
Is it not enough to simply use `activeStep`?

Reply:
This is to sync up when `activeStep` changes due to next action being done, it will auto focus next action in the checklist. It actually provides good UX.